src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -99,20 +118,20 @@ private function replaceConstructor(ClassMethod $method) : void
         // Create a set of closures that will be called to hydrate the object.
         // Array of closures in a naturally indexed array, ordered, which will
         // then be called in order in the hydrate() and extract() methods.
-        foreach ($this->hiddenPropertyMap as $className => $propertyNames) {
+        foreach ($this->hiddenPropertyMap as $className => $properties) {
             // Hydrate closures
             $bodyParts[] = '$this->hydrateCallbacks[] = \\Closure::bind(static function ($object, $values) {';
-            foreach ($propertyNames as $propertyName) {
-                $bodyParts[] = "    if (isset(\$values['" . $propertyName . "']) || " .
-                '$object->' . $propertyName . " !== null && \\array_key_exists('" . $propertyName . "', \$values)) {";
-                $bodyParts[] = '        $object->' . $propertyName . " = \$values['" . $propertyName . "'];";
-                $bodyParts[] = '    }';
+            foreach ($properties as $property) {
+                \assert($property instanceof ObjectProperty);
```

Comment:
These assertions can be dropped once the property is documented correctly

Reply:
That's true, but some IDE's miss it (sometime mine does, sometime it doesn't) - that's why I tend to add them when it shouldn't. But OK, I'll remove it.

Reply:
For me, the baseline type reference is @psalm - IDEs will indeed need to catch up :P

Reply:
Anything that can de-ambiguate the type while reading the code is good to me, I like to over-assert() sometime :)